saleor/account/views.py

```diff
@@ -156,3 +158,20 @@ def account_delete_confirm(request, token):
 
     return TemplateResponse(
         request, 'account/account_delete_prompt.html')
+
+
+@login_required
+def email_edit(request):
+    form = EmailChangeForm(request, data=request.POST or None)
+
+    if request.method == 'POST':
+        if form.is_valid():
+            form.clean_email()
+            messages.success(request, pgettext(
+                'Storefront message', 'Email successfully changed.'))
+        elif form.errors:
+            messages.error(request, pgettext(
```

Comment:
I think we should stick to the form's errors, trying to attach the error message to the most precise item possible. 

form's field -> whole form (ValidationError) -> messages